lib/writer/block-write-stream.js

```diff
@@ -134,10 +157,24 @@ class BlockWriteStream extends stream.Writable {
     }
 
     fs.write(this.fd, chunk, 0, chunk.length, chunk.position, (error, bytesWritten) => {
```

Comment:
If you fancy testing this, you can add the following here (also try EIO for a different error message):
```js
// Simulate a 80% failure rate
if (!error && (Math.random() > 0.2)) {
  error = new Error('Random I/O error')
  error.code = 'ENXIO'
}
```